tests/test_parser.py

```diff
@@ -0,0 +1,41 @@
+import subprocess
+from pathlib import Path
+
+import pytest
+import pytest_subtests
+
+
+base_directory = Path(__file__).parent.resolve()
+
+glsl_extensions = (
+    ".glsl",  ".vert",  ".frag",  ".geom",
+    ".comp",  ".tesc",  ".tese",  ".rgen",
+    ".rint",  ".rahit", ".rchit", ".rmiss",
+    ".rcall", ".mesh",  ".task"
+)
+
+
+@pytest.fixture(
+    params=("glsl-samples/well-formed/glslang/",)
+)
+def dir_with_test_files(request) -> Path:
+    return base_directory / Path(request.param)
+
+
+def test_parser_in_directory(
+    subtests: pytest_subtests.SubTests,
+    dir_with_test_files: Path
+):
+
+    for file in dir_with_test_files.iterdir():
+        if file.suffix not in glsl_extensions:
+            continue
+
+        with subtests.test(msg=str(file)):
+            output = subprocess.run(
+                args=("glsl_analyzer", "--parse-file", str(file)),
+                capture_output=True,
+                text=True
+            )
+
+            assert len(output.stderr) == 0, output.stderr
```

Comment:
Maybe check the exit code here as well (just in case)? Should be `1` on error.

Reply:
Yep, that was pretty dumb of me. Completely forgot about crashes. Thanks, will do.

Reply:
Was it the case that the `--parse-file` would return exit code 0 previously on errors? Because I have no clue why the shell script used stderr output and not the exit code for tests. Maybe it was me just being dumb, hmm.

Reply:
Now that I think about it, it was probably just me shortcutting it, since both errors and crashes write to the stderr. Anyways, I added a check for the return code.

Reply:
You are right that stderr would likely catch 99.9% of all errors, but status code should catch the remaining 0.1% 🙂